src/expr/src/linear.rs

```diff
@@ -1863,22 +1817,16 @@ pub mod plan {
 
             // Produce an output only if the upper bound exceeds the lower bound,
             // and if we did not encounter a `null` in our evaluation.
-            if lower_bound != upper_bound && !null_eval {
+            if Some(lower_bound) != upper_bound && !null_eval {
                 row_builder
                     .packer()
                     .extend(self.mfp.mfp.projection.iter().map(|c| datums[*c]));
-                let (lower_opt, upper_opt) = match (lower_bound, upper_bound) {
-                    (Some(lower_bound), Some(upper_bound)) => (
+                let (lower_opt, upper_opt) = match upper_bound {
+                    Some(upper_bound) => (
                         Some(Ok((row_builder.clone(), lower_bound, diff))),
                         Some(Ok((row_builder.clone(), upper_bound, -diff))),
                     ),
-                    (Some(lower_bound), None) => {
-                        (Some(Ok((row_builder.clone(), lower_bound, diff))), None)
-                    }
-                    (None, Some(upper_bound)) => {
-                        (None, Some(Ok((row_builder.clone(), upper_bound, -diff))))
-                    }
-                    _ => (None, None),
+                    None => (Some(Ok((row_builder.clone(), lower_bound, diff))), None),
```

Comment:
I think this can be simplified; it previously existed because cloning `row_builder` was expensive, but we seem to do that unconditionally now (and before the edit). Roughly, we always produce the `lower_bound` output, and optionally an `upper_bound` output.

I'm happy to make that edit in follow-up work.